akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -783,7 +784,8 @@ import akka.util.ccompat.JavaConverters._
       val failure = ActorSubscriberMessage.OnError(ex)
       if (!status.compareAndSet(null, failure))
         status.get.asInstanceOf[AsyncCallback[Any]].invoke(failure)
-    case OnError(_: SubscriptionTimeoutException) => // already timed out, keep the timeout as that happened first
+    case ActorSubscriberMessage.OnComplete => // it was already completed, ignore failure as completion happened first
+    case OnError(_)                        => // already failed out, keep the exception as that happened first
```

Comment:
We could still think about whether it should be an error to complete/fail twice (the previous match errors in that case were ugly definitely).

Reply:
Yeah, I noted that elsewhere, so this was mostly to make the HTTP2 tests pass, but what failed was the logic in place to avoid leaking substreams there, and that didn't look at the substream to see if it is closed before closing which may be considered a bug.

Otoh I wonder if there isn't a race here that could make it fail without this as well, if the materialized up or downstream completes or fails at roughly the same time as the stage itself autocloses the substreams, and maybe is nice if that does not cause MatchErrors.

Reply:
Yeah, I guess I'm fine with it as it is.